isis/tests/FunctionalTestsMimap2isis.cpp

```diff
@@ -545,6 +545,234 @@ TEST(FunctionalTestMimap2Isis, L3C) {
   EXPECT_PRED_FORMAT2(AssertQStringsEqual, bandbin.findKeyword("BaseBand"), "MV5");
 }
 
+TEST(FunctionalTestMimap2Isis, MAPv3) {
+   QTemporaryDir prefix;
+   QString cubeFileName = prefix.path() + "/mimap2isisTEMP.cub";
+   QVector<QString> args = {"from=data/mimap2isis/MI_MAP_03_N51E124N50E125SC_cropped.lbl",
+                            "image=data/mimap2isis/MI_MAP_03_N51E124N50E125SC_cropped.img",
+                            "to=" + cubeFileName};
+
+  UserInterface options(APP_XML, args);
+  try {
+   mimap2isis(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to ingest MI MAP image: " << e.toString().toStdString().c_str() << std::endl;
+  }
+
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+
+  // Archive Group
+  PvlGroup &archive = isisLabel->findGroup("Archive", Pvl::Traverse);
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("SoftwareName"), "RGC_TC_MI_PLUS");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("SoftwareVersion"), "4.0.0");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("ProcessVersionId"), "MAP");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("ProductCreationTime"), "2015-01-02T01:45:16");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("ProgramStartTime"), "2015-01-02T01:43:13");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("ProducerId"), "LISM");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("ProductSetId"), "MI_MAP");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("ProductVersionId"), "03");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("RegisteredProduct"), "Y");
+
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("SpiceMetakernelFileName"), "RGC_INF_TCv401IK_MIv200IK_SPv105IK_RISE100i_05_100h_02_LongCK_DS_V02_de421_131210.mk");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("DataSetId"), "SLN-L-MI-5-MAP-V3.0");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("ImageValueType"), "REFLECTANCE");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, archive.findKeyword("ImageUnit"), "ND");
+
+  int i;
```

Comment:
Don't define the variable you are iterating on outside the for loop unless you need to. It can cause weird errors that are hard to track down. It's probably okay here, but is generally poor practice.

Reply:
This was something that got copied from the other tests so I also fixed it in the other two tests.